File: php/src/PendantNecklace.php
<?php

namespace NecklaceRefactoringKata;

class PendantNecklace extends Necklace
{
    public function __construct(
        public Jewel $stone,
        public NecklaceType $type,
        public Necklace $chain,
        public Jewellery $pendant
    ) {
    }

    public function isLarge(): bool
    {
        return $this->chain->isLarge() || $this->pendant->isLarge();
    }
}

File: kotlin/src/main/kotlin/org/sammancoaching/necklace/PendantNecklace.kt
package org.sammancoaching.necklace

data class PendantNecklace(
  val chain: Necklace,
  val pendant: Jewellery,
  override val type: NecklaceType,
  override val stone: Jewel,
) : Necklace(type, stone) {
  override fun isLarge(): Boolean =
    chain.isLarge() || pendant.isLarge()
}
